### PR TITLE
fix: handle opening external links when using mouse scroll click

### PR DESCRIPTION
**Pull request by alfonsobries**
## Summary

As the title says

https://app.clickup.com/t/yy957w

To test: merge this on msq and try to open an external link with the mouse scroll

## Checklist

-   [x] I checked my UI changes against the design and there are no notable differences
-   [x] I checked my UI changes for any responsiveness issues
-   [x] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I provided a screenshot of my changes to the component _(if applicable)_
-   [ ] I regenerated the `icons.html` file and checked if my newly added icon is shown correctly _(if necessary)_
-   [ ] I added an explanation on how to use the component to the readme _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

**Review by ItsANameToo — CHANGES_REQUESTED** [review]
not working when used on msq with `yarn prod`

File: resources/views/external-link-confirm.blade.php [hunk 1]
```diff
@@ -100,14 +100,24 @@ function initExternalLinkConfirm() {
...
-            link.addEventListener('click', function(e) {
+            const clickHandler = (e) => {
                 if (localStorage.getItem('has_disabled_link_warning') === 'true') {
                     return;
                 }
 
                 e.preventDefault();
 
                 Livewire.emit('openModal', 'external-link-confirm', link.getAttribute('href'));
+            }
```
Comment (ItsANameToo) [inline resources/views/external-link-confirm.blade.php:111]: this probably needs a `;` to avoid issues when minimized

**Review by ItsANameToo — CHANGES_REQUESTED** [review]

File: resources/views/external-link-confirm.blade.php [hunk 2]
```diff
@@ -100,14 +100,24 @@ function initExternalLinkConfirm() {
...
+
+            link.addEventListener('click', clickHandler);
+
+            link.addEventListener('mousedown', (e) => {
+                // Meaning the user clicked the middle mouse button (scroll).
+                // The rest of the buttons are handled bu the `click` event.
```
Comment (ItsANameToo) [inline resources/views/external-link-confirm.blade.php:117]:
```suggestion
                // The rest of the buttons are handled by the `click` event.
```

File: resources/views/external-link-confirm.blade.php [hunk 3]
```diff
@@ -100,14 +100,24 @@ function initExternalLinkConfirm() {
...
                 }
 
                 e.preventDefault();
 
                 Livewire.emit('openModal', 'external-link-confirm', link.getAttribute('href'));
+            }
+
+            link.addEventListener('click', clickHandler);
+
+            link.addEventListener('mousedown', (e) => {
```
Comment (ItsANameToo) [inline resources/views/external-link-confirm.blade.php:115]: issue with `mousedown` is that it triggers immediately, and not after "clicking". Wouldn't something like https://developer.mozilla.org/en-US/docs/Web/API/Element/auxclick_event be more suitable to keep it in line with the regular click handling? 🤔

Comment (ItsANameToo) [reply resources/views/external-link-confirm.blade.php:115]: nvm no full support for that event in all browsers :sadup:

still think it feels a bit odd having the middle click open it immediately as opposed to the regular click, will check for alternatives

**Comment by alfonsobries** [comment]
@ItsANameToo the auxclick event did the trick, the docs say that it not works on safari but tested and it works fine (maybe doesnt work on safari but safari handles the middle click differently)

Tested of firefox, brave safari  and Microsoft edget (for mac)

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**